commit d9b8808327e9a7f28b4129373057aaf4b9d6856f
Author: Pavan Balaji <balaji@mcs.anl.gov>
Date:   Fri Jan 1 20:15:16 2010 -0600

    Hack fix for the include-path passing problem in dllchan (see ticket

diff --git a/src/mpid/ch3/channels/shm/configure.in b/src/mpid/ch3/channels/shm/configure.in
index 3c289dd..af53eb5 100644
--- a/src/mpid/ch3/channels/shm/configure.in
+++ b/src/mpid/ch3/channels/shm/configure.in
@@ -181,10 +181,12 @@ export ETAGSADD
 
 PAC_UPDATE_BASE_CACHE
 
-# Add the MPICH2 include flags to CPPFLAGS.  See the discussion above about
-# which include flags should start in MPICH2_INCLUDE_FLAGS and which in 
-# CPPFLAGS
-CPPFLAGS="$CPPFLAGS $MPICH2_INCLUDE_FLAGS"
+# Add the MPICH2 include flags to CPPFLAGS. Add the
+# MPICH2_INCLUDE_FLAGS to the front. This is a hack for dllchan to
+# work, since it adds its own include paths before the channel can add
+# the appropriate include paths; we need to make sure our include
+# paths are to the front.
+CPPFLAGS="$MPICH2_INCLUDE_FLAGS $CPPFLAGS"
 
 # define the ABI version of the channel ABI that we are using.
 ABIVERSION=1:1
diff --git a/src/mpid/ch3/channels/sock/configure.in b/src/mpid/ch3/channels/sock/configure.in
index 4e5c4c9..13d59ca 100644
--- a/src/mpid/ch3/channels/sock/configure.in
+++ b/src/mpid/ch3/channels/sock/configure.in
@@ -129,10 +129,12 @@ export ETAGSADD
 
 PAC_UPDATE_BASE_CACHE
 
-# Add the MPICH2 include flags to CPPFLAGS.  See the discussion above about
-# which include flags should start in MPICH2_INCLUDE_FLAGS and which in 
-# CPPFLAGS
-CPPFLAGS="$CPPFLAGS $MPICH2_INCLUDE_FLAGS"
+# Add the MPICH2 include flags to CPPFLAGS. Add the
+# MPICH2_INCLUDE_FLAGS to the front. This is a hack for dllchan to
+# work, since it adds its own include paths before the channel can add
+# the appropriate include paths; we need to make sure our include
+# paths are to the front.
+CPPFLAGS="$MPICH2_INCLUDE_FLAGS $CPPFLAGS"
 
 # define the ABI version of the channel ABI that we are using.
 ABIVERSION=1:1
diff --git a/src/mpid/ch3/channels/ssm/configure.in b/src/mpid/ch3/channels/ssm/configure.in
index 66b42f3..f3a9d2c 100644
--- a/src/mpid/ch3/channels/ssm/configure.in
+++ b/src/mpid/ch3/channels/ssm/configure.in
@@ -261,10 +261,12 @@ export ETAGSADD
 
 PAC_UPDATE_BASE_CACHE
 
-# Add the MPICH2 include flags to CPPFLAGS.  See the discussion above about
-# which include flags should start in MPICH2_INCLUDE_FLAGS and which in 
-# CPPFLAGS
-CPPFLAGS="$CPPFLAGS $MPICH2_INCLUDE_FLAGS"
+# Add the MPICH2 include flags to CPPFLAGS. Add the
+# MPICH2_INCLUDE_FLAGS to the front. This is a hack for dllchan to
+# work, since it adds its own include paths before the channel can add
+# the appropriate include paths; we need to make sure our include
+# paths are to the front.
+CPPFLAGS="$MPICH2_INCLUDE_FLAGS $CPPFLAGS"
 
 dnl Add a step to create the dependencies directory if it is not present
 AC_OUTPUT_COMMANDS([if grep 'dependencies:' Makefile >/dev/null 2>&1 ; then
